### PR TITLE
DLS-10052 - Play 30 upgrade + bootstrap upgrade + optimisations

### PR DESCRIPTION
**Pull request by TomRafferty**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -21,16 +21,16 @@ import sbt.*
 object AppDependencies {
 
   val bootstrapVersion         = "8.4.0"
-  val playVersion               = "play-28"
-  val playPartialsVersion      = s"8.4.0-$playVersion"
-  val hmrcMongoVersion         = "1.3.0"
+  val playVersion               = "play-30"
+  val playPartialsVersion      = "9.1.0"
```
Comment (karolina-zaborska) [inline project/AppDependencies.scala:24]: Just an observation: I don't see a point in having `val playPartialsVersion` as that's used only once

File: project/plugins.sbt [hunk 2]
```diff
@@ -5,10 +5,12 @@ addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.20.0")
 
 addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.2.0")
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.20")
+addSbtPlugin("org.playframework" %% "sbt-plugin" % "3.0.1")
```
Comment (karolina-zaborska) [inline project/plugins.sbt]: doesn't have to be, but could be `3.0.2`

File: build.sbt [hunk 3]
```diff
@@ -48,7 +48,6 @@ lazy val microservice = Project(appName, file("."))
   .configs(IntegrationTest)
   .settings(inConfig(IntegrationTest)(Defaults.itSettings): _*)
   .settings(resolvers += Resolver.jcenterRepo)
-  .settings(integrationTestSettings())
```
Comment (karolina-zaborska) [inline build.sbt:51]: We removed that in stubs, should we remove it in frontend?

I thought that in frontends we follow the migration guide and replace it with:
`.configs(ItTest)` & `.settings(inConfig(ItTest)(Defaults.testSettings): _*)` and so on

File: project/AppDependencies.scala (hunk 1 above)
Comment (TomRafferty) [reply project/AppDependencies.scala:24]: I noticed this as well, decided to leave it in as it was added by a talented frontend dev and I didn't want to go over his work as it's functional anyway - but yes I agree that it seems redundant

File: build.sbt (hunk 3 above)
Comment (TomRafferty) [reply build.sbt:51]: I can try out that replacement - didn't see that part of the guide

**Review by BetaDraconis — APPROVED** [review]

**Pull request merged**